xapian-applications/omega/omega.cc

```diff
@@ -552,12 +593,21 @@ try {
 	} while (*p);
 
 	val = cgi_params.find("SORTREVERSE");
+	if (val != cgi_params.end() && (val->second[0] < '0' ||
+	    val->second[0] > '9' || val->second[0] == '-')) {
+	    cerr << "Value entered is not in range\n"
+		<< "Range is 0-1";
+	}
 	if (val != cgi_params.end() && atoi(val->second.c_str()) != 0) {
 	    reverse_sort = !reverse_sort;
 	}
-
 	val = cgi_params.find("SORTAFTER");
 	if (val != cgi_params.end()) {
+	    if (val->second[0] < '0' || val->second[0] > '9' ||
+		val->second[0] == '-') {
+		cerr << "Value entered is not in range\n"
+		    << "Range is 0-1";
```

Comment:
The documentation for both `SORTREVERSE` and `SORTAFTER` says "if non-zero, [has some effect]", not "must be 0 or 1".

Reply:
thanks i will fix that.